tests/span_tests.cpp

```diff
@@ -1297,3 +1300,13 @@ TEST(span_test, front_back)
     EXPECT_DEATH(s2.front(), expected);
     EXPECT_DEATH(s2.back(), expected);
 }
+
+#if defined(FORCE_STD_SPAN_TESTS) || defined(__cpp_lib_span) && __cpp_lib_span >= 202002L
+TEST(span_test, std_span)
+{
+    // make sure std::span can be constructed from gsl::span
+    gsl::span<int> gsl_span;
+    std::span<int> std_span = gsl_span;
+    (void)std_span; // suppress unused variable warning
```

Comment:
I recommend:
```suggestion
    int arr[5] = {1, 2, 3, 4, 5};
    gsl::span<int> gsl_span{arr};
#ifdef __cpp_lib_ranges
    EXPECT_TRUE(std::to_address(gsl_span.begin()) == gsl_span.data());
    EXPECT_TRUE(std::to_address(gsl_span.end()) == gsl_span.data() + gsl_span.size());
#endif // __cpp_lib_ranges

    std::span<int> std_span = gsl_span;
    EXPECT_TRUE(std_span.data() == gsl_span.data());
    EXPECT_TRUE(std_span.size() == gsl_span.size());
```
which will fail at `EXPECT_TRUE(std::to_address(gsl_span.end()) == gsl_span.data() + gsl_span.size());` since `operator->` isn't well-defined for the end iterator of a span. (The `contiguous_iterator` requires `std::to_address(meow)` to work, which defaults to returning `operator->()`.)

I assume you don't want `operator->` to work for past-the-end iterators any more than we do, so you'll need to specialize `std::pointer_traits` and implement `to_address` to tell `std::to_address` how to convert your iterators into pointers. 

Actually, it would be easier to make a PR into your branch than type this up. I've submitted https://github.com/dmitrykobets-msft/GSL/pull/1.